test.py

```diff
@@ -1,99 +1,266 @@
+
+    @mock.patch("carapp.db_models.Car")
+    @mock.patch("carapp.db.session")
+    def test_post_car(self, MockSession, MockCar):
+        CAR = {
+            "id": 100000,
+            "car_title": "Supercar",
+            "title": "Supercar_3000",
+            "condition": "New",
+            "body_type": "Sedan",
+            "brand": "Supercar",
+            "model": "Car",
+            "year_of_production": "3000",
+            "country_origin": "Supercountry",
+            "country_now": "Supercountry",
+            "city": "Supercity",
+            "mileage": "0 km",
+            "technical_condition": "Supercondition",
+            "gear_box": "Manual",
+            "fuel_type": "Gasoline",
+            "engine": "10 l (666 hp / 100 kW)",
+            "color": "White",
+            "safety": "Supersafe",
+            "wanted": "No",
+            "multimedia": "Subwoofer",
+            "comfort": "Supercomfort",
+            "picture": "Cool pic",
+            "price": "1000000$",
+            "warranty": "12 months",
+            "fuel_consumption": "2l/100km",
+            "drivetrain": "4matic",
+            "date_added": "2020-10-21 15:02:54.770364",
+            "user_id": 999,
+        }
+        with app.test_client() as client:
+            MockCar.return_value = CAR
+            response = client.post("/car", json=CAR)
+            self.assertIsNotNone(response.get_json())
+            self.assertIsInstance(response.get_json(), dict)
+            self.assertEqual(response.status_code, 201)
+            self.assertEqual(response.get_json(), CAR)
+
+    @mock.patch("carapp.db_models.Car")
+    @mock.patch("carapp.db.session")
+    def test_put_car(self, MockSession, MockCar):
+        CAR = {
+            "id": 100000,
+            "car_title": "Supercar",
+            "title": "Supercar 3000",
+            "condition": "New",
+            "body_type": "Sedan",
+            "brand": "Supercar",
+            "model": "Car",
+            "year_of_production": "3000",
+            "country_origin": "Supercountry",
+            "country_now": "Supercountry",
+            "city": "Supercity",
+            "mileage": "0 km",
+            "technical_condition": "Supercondition",
+            "gear_box": "Manual",
+            "fuel_type": "Gasoline",
+            "engine": "10 l (666 hp / 100 kW)",
+            "color": "White",
+            "safety": "Supersafe",
+            "wanted": "No",
+            "multimedia": "Subwoofer",
+            "comfort": "Supercomfort",
+            "picture": "Cool pic",
+            "price": "1000000$",
+            "warranty": "12 months",
+            "fuel_consumption": "2l/100km",
+            "drivetrain": "4matic",
+            "date_added": "2020-10-21 15:02:54.770364",
+            "user_id": 999,
+        }
+        with app.test_client() as client:
+            MockCar.query.filter_by.update.return_value = CAR
+            response = client.put("/car/100000", json=CAR)
```

Comment:
Put a breakpoint here. What is the reason for the 422 response code? This is definitely about your payload.

Reply:
It is not a bad solution to pass id in the path parameters. Do you think your payload is consistent with the expectations of the endpoint?

Reply:
Ok, I have found a problem in **test.py** and **scheme.py** already works good :)